authorization.go

```diff
@@ -98,30 +99,43 @@ func authorize(r *http.Request, jwtConfig *jwtConfig, publishOrigins []string, c
 
 	for _, allowedOrigin := range publishOrigins {
 		if allowedOrigin == "*" || origin == allowedOrigin {
-			return validateJWT(cookie.Value, jwtConfig)
+			return validateJWT(cookie.Value, jwtConfig, jwks)
 		}
 	}
 
 	return nil, fmt.Errorf("%q: %w", origin, ErrOriginNotAllowed)
 }
 
 // validateJWT validates that the provided JWT token is a valid Mercure token.
-func validateJWT(encodedToken string, jwtConfig *jwtConfig) (*claims, error) {
-	token, err := jwt.ParseWithClaims(encodedToken, &claims{}, func(token *jwt.Token) (interface{}, error) {
-		switch jwtConfig.signingMethod.(type) {
-		case *jwt.SigningMethodHMAC:
-			return jwtConfig.key, nil
-		case *jwt.SigningMethodRSA:
-			pub, err := jwt.ParseRSAPublicKeyFromPEM(jwtConfig.key)
-			if err != nil {
-				return nil, fmt.Errorf("unable to parse RSA public key: %w", err)
+func validateJWT(encodedToken string, jwtConfig *jwtConfig, jwksURL string) (*claims, error) {
+	var keyFunc jwt.Keyfunc
+
+	if jwksURL != "" {
```

Comment:
nitpicking: I would invert the test to avoid the negation as there is an `else`.

Reply:
The reason I did that is to prioritize jwks when both jwks and jwt config are present. Do you see a way we could combine these?